cmd/permissions-checker/permissionschecks.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/user"
+	"path/filepath"
+	"regexp"
+	"strings"
+	"syscall"
+)
+
+// FindRecursive iterate in a directory (path) showing permissions in a recursive way
+/* path  - where to look for files and directories
+ * level - used to print the output in a hierarchical way
+ */
+func FindRecursive(path string, level int) {
+	files, err := ioutil.ReadDir(path)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// In verbose mode we are printing the output in a hierarchical way
+	if Verbose {
+		level++
+	}
+
+	for _, f := range files {
+		name := f.Name()
+
+		relativePath := strings.TrimPrefix(strings.Join([]string{path, name}, "/"), Directory)
+		fullPath := strings.Join([]string{path, name}, "/")
+
+		if m, _ := regexp.MatchString(Exclude, relativePath); m {
+			if Verbose {
+				fmt.Printf("\x1b[33m%sExcluding %s\n\x1b[0m", strings.Repeat(" ", level), name)
+			}
+		} else {
+			if ShowHidden || !strings.HasPrefix(name, ".") {
+				mode := f.Mode()
+				currentOwner, _ := user.LookupId(fmt.Sprint(f.Sys().(*syscall.Stat_t).Uid))
+				currentGroup, _ := user.LookupGroupId(fmt.Sprint(f.Sys().(*syscall.Stat_t).Gid))
+				currentPerm := mode.Perm().String()
+				hasCorrectOwner := currentOwner.Username == DefaultOwner
+				hasCorrectGroup := currentGroup.Name == DefaultGroup
+
+				if mode.IsRegular() {
+					hasCorrectPermissions := checkPermissions(currentPerm, DefaultFilePerm)
+					printOutput(level, "f", fullPath, currentPerm, DefaultFilePerm, currentOwner.Username, currentGroup.Name, hasCorrectPermissions, hasCorrectOwner, hasCorrectGroup)
+				} else if mode.IsDir() {
+					hasCorrectPermissions := checkPermissions(currentPerm, DefaultDirPerm)
+					printOutput(level, "d", fullPath, currentPerm, DefaultDirPerm, currentOwner.Username, currentGroup.Name, hasCorrectPermissions, hasCorrectOwner, hasCorrectGroup)
+					FindRecursive(fullPath, level)
+				}
+			}
+		}
+	}
+}
+
+// checkPermissions return true if the permissions are correct (false in other case)
```

Comment:
Use the third person: checkPermissions `returns` true... Apply to the rest of the functions

Source: https://blog.golang.org/godoc-documenting-go-code